server/pulp/server/content/web/views.py

```diff
@@ -128,6 +127,13 @@ def redirect(request, key):
     def __init__(self, **kwargs):
         super(ContentView, self).__init__(**kwargs)
         self.key = Key.load(pulp_conf.get('authentication', 'rsa_key'))
+        # Make sure all requested paths fall under these sub-directories, otherwise
+        # we might find ourselves serving private keys to all and sundry.
+        self.safe_serving_paths = [os.path.realpath(os.path.join(LOCAL_STORAGE, subdir))
```

Comment:
Should use `config.get('server', 'storage_dir')` instead of `constants.LOCAL_STORAGE`.  In fact, `constants.LOCAL_STORAGE` should not exist but that's outside the scope of this PR.